src/index.ts

```diff
@@ -12,23 +12,27 @@ interface YamlPluginOptions {
   ) => object | undefined;
 }
 
+const DETECTOR_R = /\.ya?ml$/;
+
 export const yamlPlugin = (options: YamlPluginOptions): Plugin => ({
   name: "yaml",
   setup(build) {
     // resolve .yaml and .yml files
-    build.onResolve({ filter: /\.(yml|yaml)$/ }, (args) => {
+    build.onResolve({ filter: DETECTOR_R }, (args) => {
       if (args.resolveDir === "") return;
 
+      const filePath = path.isAbsolute(args.path)
+        ? args.path
+        : path.join(args.resolveDir, args.path);
+
       return {
-        path: path.isAbsolute(args.path)
-          ? args.path
-          : path.join(args.resolveDir, args.path),
-        namespace: "yaml"
+        namespace: "file",
+        path: filePath
       };
     });
 
     // load files with "yaml" namespace
```

Comment:
This comment will be out of date? It should either be removed or changed to `file`?